lib/SourceMapDevToolPlugin.js

```diff
@@ -47,13 +47,21 @@ const validate = createSchemaValidation(
  * @property {ItemCacheFacade} cacheItem cache item
  */
 
+const METACHARACTERS_REGEXP = /[-[\]\\/{}()*+?.^$|]/g;
```

Comment:
The one hazard of hoisting regexps is that they maintain internal state, so in some cases you'll need to set `.lastIndex = -1` before each time you use the regexp.

Reply:
Would you mind checking my changes here? Looks like `.test()` was only called in two places that would warrant this gaurd.